lib/pages/authentication-page.dart

```diff
@@ -0,0 +1,261 @@
+                  },
+                  label: Text(getString(context, 'google'), style: TextStyle(color: Colors.grey[600], fontWeight: FontWeight.bold),), 
+                  color: Colors.white,
+                  icon: new Image.asset('assets/images/Google_g.png', width: 20,),              
+                ),
+              ),
+              Positioned(
+                top: 460,
+                left: (MediaQuery.of(context).size.width / 2) - 100,
+                child : RaisedButton.icon(
+                  onPressed: () async { 
+                    
+                  },
+                  label: Text(getString(context, 'facebook'), style: TextStyle(color: Colors.grey[600], fontWeight: FontWeight.bold),), 
+                  color: Colors.white,
+                  icon: new Image.asset('assets/images/facebook_f.png', width: 20,),              
+                ),
+              ),
+              Positioned(
+                top: 510,
+                left: (MediaQuery.of(context).size.width / 2) - 90,
+                child : RaisedButton.icon(
+                  onPressed: () async { 
+                    
+                  },
+                  label: Text(getString(context, 'twitter'), style: TextStyle(color: Colors.grey[600], fontWeight: FontWeight.bold),), 
+                  color: Colors.white,
+                  icon: new Image.asset('assets/images/twitter.png', width: 20,),              
+                ),
+              ),
+              Positioned(
+                key: Key("register_page_button"),
+                right: 10,
+                bottom: 10,
+                child: FlatButton(
+                  color: Colors.transparent,
+                  onPressed: () => openRegisterPage(context),
+                  child: Text(getString(context, 'create_account'), 
+                    style: TextStyle(
+                      color: Colors.blue,
+                    ),
+                  ),
+                ),
+              ),
+              Positioned(
+                key: Key("forgotten_password_button"),
+                left: 10,
+                bottom: 10,
+                child: FlatButton(
+                  color: Colors.transparent,
+                  onPressed: () => null,
+                  child: Text(getString(context, 'forgotten_password'), 
+                    style: TextStyle(
+                      color: Colors.blue,
+                    ),
+                  ),
+                ),
+              ),
+            ],
+          ), 
+        ), 
+      ),
+    );
+  }
+
+  Future<void> signInUser() async {
+    try {
+      UserService().user = await AuthenticationService().signIn(_email, _password);
+      if (UserService().user != null) {
+        openMainPage(context);
+        Fluttertoast.showToast(msg: getString(context, 'signed_in'));
+      }
+    } catch (e) {
+      debugPrint(e.message);
+      switch (e.message) {
+        case "The password is invalid or the user does not have a password.":
```

Comment:
never make a switch case on a translation value. If Firebase changes the text your code will no longer work